test/assertions.js

```diff
@@ -35,13 +75,155 @@ should.Assertion.add('hidden', function hidden(computed = false) {
+  // inserting elements into the DOM. (That happens with v-tooltip, for
+  // example.) That's why we fall back to document.getElementById().
+  const describer = element.getRootNode().querySelector(`#${id}`) ??
+    document.getElementById(id);
+  should.exist(describer);
+  const text = describer.textContent.trim();
+  if (description != null)
+    text.should.match(description);
+  else
+    text.should.not.equal('');
+});
+
+Assertion.add('ariaDescriptions', function ariaDescriptions(descriptions = undefined) {
+  this.params = { operator: 'to be described by multiple elements' };
+  const element = unwrapElement(this.obj);
+  const describedBy = element.getAttribute('aria-describedby');
+  should.exist(describedBy);
+  const ids = describedBy.split(' ');
+  ids.length.should.be.above(1);
+  const text = ids.map(id => {
+    // See above for why we fall back to document.getElementById().
+    const describer = element.getRootNode().querySelector(`#${id}`) ??
+      document.getElementById(id);
+    should.exist(describer);
+    return describer.textContent.trim();
+  });
+  if (descriptions != null) {
+    text.length.should.equal(descriptions.length);
+    for (let i = 0; i < text.length; i += 1)
+      text[i].should.match(descriptions[i]);
+  } else {
+    for (const s of text) s.should.not.equal('');
+  }
+});
+
+// Asserts that an element shows a tooltip on mouseenter and hides it on
+// mouseleave. If `text` is specified, asserts that the text of the tooltip
+// matches `text`.
+Assertion.addAsync('tooltip', async function tooltip(text = undefined) {
+  this.params = { operator: 'to have a tooltip' };
+  document.querySelectorAll('.tooltip.in').length.should.equal(0);
+  const element = unwrapElement(this.obj);
+  element.dispatchEvent(new MouseEvent('mouseenter'));
+  await wait();
+  const tooltips = document.querySelectorAll('.tooltip.in');
+  tooltips.length.should.equal(1);
+  const actualText = tooltips[0].textContent.trim();
+  if (text != null)
+    actualText.should.match(text);
+  else
+    actualText.should.not.equal('');
+  element.dispatchEvent(new MouseEvent('mouseleave'));
+  document.querySelectorAll('.tooltip.in').length.should.equal(0);
+});
+
+/* Asserts that an element shows the entirety of its text in a tooltip if and
+only if its text is truncated. The assertion works by replacing the text with a
+long string, checking that that shows a tooltip, then replacing the text with a
+short string and checking that that does not show a tooltip. If the element does
+not truncate its text (by setting text-overflow) and does not have an ancestor
+that truncates its text, the assertion will try to insert a new ancestor into
+the tree to truncate its text. In other words, the assertion does not test that
+the element has truncated text, only that if it has truncated text, it will show
+that text in a tooltip. */
+Assertion.addAsync('textTooltip', async function textTooltip() {
```

Comment:
I have mixed feelings about this assertion. It's used in a lot of tests as a replacement for assertions about the `title` attribute. The assertion in the test itself is simple to use. However, under the hood, things are pretty complicated here. Given how simple `v-tooltip.text` is, part of me wonders how important it is to be verifying these tooltips. Maybe this assertion isn't really needed? I guess we have it, and it's not likely to change, so we might as well keep it. 🤔